mesonbuild/dependencies/misc.py

```diff
@@ -33,6 +32,50 @@
 )
 
 
+class HDF5Dependency(ExternalDependency):
+
+    def __init__(self, environment, kwargs):
+        language = kwargs.get('language', 'c')
+        super().__init__('hdf5', environment, language, kwargs)
+        kwargs['required'] = False
+        kwargs['silent'] = True
+        self.is_found = False
+
+        pkgconfig_files = ['hdf5']
+
+        if language not in ('c', 'cpp', 'fortran'):
+            raise DependencyException('Language {} is not supported with HDF5.'.format(language))
+
+        for pkg in pkgconfig_files:
+            try:
+                pkgdep = PkgConfigDependency(pkg, environment, kwargs, language=self.language)
+                if pkgdep.found():
+                    self.compile_args = pkgdep.get_compile_args()
+                    # derive needed libraries by language
+                    link_args = pkgdep.get_link_args()
+                    lang_link_args = []
+                    for larg in link_args:
+                        lpath = Path(larg)
+                        if lpath.is_file():                            
+                            if language == 'cpp':
+                                lang_link_args.append(str(lpath.parent / (lpath.stem + '_hl_cpp' + lpath.suffix)))
+                                lang_link_args.append(str(lpath.parent / (lpath.stem + '_cpp' + lpath.suffix)))
+                            elif language == 'fortran':
+                                lang_link_args.append(str(lpath.parent / (lpath.stem + 'hl_fortran' + lpath.suffix)))
+                                lang_link_args.append(str(lpath.parent / (lpath.stem + '_fortran' + lpath.suffix)))
+                            
+                             # C is used by other languages
```

Comment:
__[Flake8]__ [E114] indentation is not a multiple of four (comment)

<sub>posted by Sider</sub>